lib/common/grpc-service.js

```diff
@@ -343,6 +344,66 @@ GrpcService.prototype.requestStream = function(protoOpts, reqOpts) {
 };
 
 /**
+ * Make an authenticated writable streaming request with gRPC.
+ *
+ * @param {object} protoOpts - The proto options.
+ * @param {string} protoOpts.service - The service git stat.
+ * @param {string} protoOpts.method - The method name.
+ * @param {number=} protoOpts.timeout - After how many milliseconds should the
+ *     request cancel.
+ * @param {object} reqOpts - The request options.
+ */
+GrpcService.prototype.requestWritableStream = function(protoOpts, reqOpts) {
+  var stream = protoOpts.stream = protoOpts.stream || duplexify.obj();
+
+  if (global.GCLOUD_SANDBOX_ENV) {
+    return stream;
+  }
+
+  var self = this;
+
+  if (!this.grpcCredentials) {
+    // We must establish an authClient to give to grpc.
+    this.getGrpcCredentials_(function(err, credentials) {
+      if (err) {
+        setImmediate(function() {
+          stream.destroy(err);
+        });
+        return;
+      }
+
+      self.grpcCredentials = credentials;
+      self.requestWritableStream(protoOpts, reqOpts);
+    });
+
+    return stream;
+  }
+
+  var service = this.getService_(protoOpts);
+  var grpcOpts = {};
+
+  if (is.number(protoOpts.timeout)) {
+    grpcOpts.deadline = GrpcService.createDeadline_(protoOpts.timeout);
+  }
+
+  var grpcStream = service[protoOpts.method](reqOpts, grpcOpts);
+  stream.setReadable(grpcStream);
+  stream.setWritable(grpcStream);
+
+  grpcStream
+    .on('status', function(status) {
```

Comment:
Yeah, this isn't an event that comes from retryRequest, if that's what you were sayin'.